core/src/main/java/org/apache/calcite/avatica/remote/looker/LookerRemoteMeta.java

```
@@ -0,0 +1,438 @@
+
+        while (parser.nextToken() != null) {
+          if (currentOffset == 0) {
+            // TODO: Handle `metadata`. We are currently ignoring it and seeking to `rows` array
+            seekToRows(parser);
+          }
+
+          int rowsRead = 0;
+          List<Object> rows = new ArrayList<>();
+
+          while (rowsRead < fetchSize) {
+            List<Object> columnValues = new ArrayList<>();
+            // the signature should _always_ have the correct number of columns.
+            // if not, something went wrong during query preparation on the Looker instance.
+            for (int i = 0; i < signature.columns.size(); i++) {
+              seekToValue(parser);
+
+              if (parser.isClosed()) {
+                // the stream is closed - all rows should be accounted for
+                currentOffset += rowsRead;
+                frameQueue.put(makeFrame(currentOffset, /*done=*/true, rows));
+                return;
+              }
+
+              // add the value to the column list
+              columnValues.add(deserializeValue(parser, signature.columns.get(i)));
+            }
+
+            // Meta.CursorFactory#deduce will select an OBJECT cursor if there is only a single
+            // column in the signature. This is intended behavior. Since the rows of a frame are
+            // simply `Object` - we could get illegal casts from an ArrayList to the underlying Rep
+            // type if we don't discard the wrapping ArrayList for the single value here.
+            Object row = signature.columns.size() == 1 ? columnValues.get(0) : columnValues;
+
+            rows.add(row);
+            rowsRead++;
+          }
+          // we fetched the allowed number of rows so add the complete frame to the queue
+          currentOffset += rowsRead;
+          frameQueue.put(makeFrame(currentOffset, /*done=*/false, rows));
+        }
+      } catch (Exception e) {
+        // enqueue all exceptions for the main thread to report
```

The comment says `exceptions` but I only see one? To be clear I think failing fast on first exception makes sense, so the comment is probably wrong.